### ISSUE TITLE
Hotswap between raw text and tree editors

### ISSUE THREAD

**Issue by Spyderrock** [bridge-core/editor#801]
**Summary**
It would be nice to be able to swap between tree and raw text editors with a macro or keypress, or at least a button on the screen. This would make it much easier to edit as I use them interchangeably. 

**Alternatives/Considerations**
Maybe having both open at the same time, though that seems harder.

**Additional context/Motivation**
This would just make it much quicker and smoother for making addons.

**Comment by solvedDev** [comment]
I agree this would be nice to have but the problem we have is that the text editor can contain invalid JSON while the tree editor requires valid JSON. We could show an information window when swapping the editor is currently not possible

**Comment by solvedDev** [comment]
I've started looking into this and I ran into more issues. There's no way we can keep history entries upon switching the editor types
